### PR TITLE
Implement a "wakeup-source" option for the i2c-rtc DeviceTree overlay.

### PR DESCRIPTION
**Pull request by nboullis**
This patch was already discussed at [https://github.com/raspberrypi/linux/pull/1260#issuecomment-204966308].
Unfortunately, I have not had the opportunity yet to test it myself with the latest firmware.

### CODE REVIEW

**Comment by pelwell** [comment]
Given that this is a parameter which will do nothing if not used, I'm prepared to take the chance that it won't work. I would, though, like an explanation of "wakeup-source" in the README.

File: arch/arm/boot/dts/overlays/i2c-rtc-overlay.dts [hunk 1]
```diff
@@ -65,5 +65,9 @@
 		pcf8523 = <&pcf8523>,"status";
 		pcf8563 = <&pcf8563>,"status";
 		trickle-resistor-ohms = <&ds1339>,"trickle-resistor-ohms:0";
+		wakeup-source = <&ds1339>,"wakeup-source?",
+				<&ds3231>,"wakeup-source?",
+				<&mcp7940x>,"wakeup-source?";
```
Comment (pelwell) [inline arch/arm/boot/dts/overlays/i2c-rtc-overlay.dts]: The semicolon on this line should be a comma.

**Comment by nboullis** [comment]
Thanks for your comment. I just pushed a fixed commit to my rpi-4.4.y branch, but can’t figure out how to update my pull request. (Sorry, still quite new to git…)

**Comment by pelwell** [comment]
The README entry contained a line which was too long (there's an 80 character limit), but I've corrected that and merged it manually.

**Comment by nboullis** [comment]
Sorry @pelwell for not figuring out the 80-character limit, and thanks for your merge.

**Pull request closed without merging**